### ISSUE TITLE
Simplify deployment to embedded platforms

### ISSUE THREAD

**Issue by mphasize** [node-opcua/node-opcua#363]
Hi there, 
this looks like a great development effort and a very nice package to get started with OPC UA.
I know that not everybody in the industry agrees that OPC UA should be run on embedded hardware in field devices, but with the OPC/UA Embedded Profile and the upcoming publisher/subscriber patterns it would be quite sexy to run node-opcua also on embedded hardware like Raspberry or the Tessel 2.

Since these environments are very constrained in disk space and memory, it would be great to simplify the `node-opcua` codebase and see if it can be refactored to use less external dependencies, especially working around big dependencies (like underscore) where the feature set is probably not fully used.

But before any work is put into this, I wanted to ask if these small microprocessor boards are meant to be a target for this package or if they are out of scope?

**Comment by mphasize** [comment]
For reference, trying to run `node-opcua` on a Tessel 2 transfers ~21MB of code to the board and ends in a `FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory` – and the Tessel has 64MB RAM.
As far as I know there's nothing inherent to the OPC/UA stack that needs that much memory, so it would be a matter of streamlining the implementation, right?
